fog/overseer/server/src/worker.rs

```diff
@@ -0,0 +1,546 @@
+            "Trying to activate an idle node with inactive outstanding key: {:?}",
+            &inactive_outstanding_key
+        );
+        for ingest_summary_node_mapping in ingest_summary_node_mappings {
+            let node_ingress_key = match CompressedRistrettoPublic::try_from(
+                ingest_summary_node_mapping
+                    .ingest_summary
+                    .get_ingress_pubkey(),
+            ) {
+                Ok(key) => key,
+                Err(_) => continue,
+            };
+            if inactive_outstanding_key.eq(&node_ingress_key) {
+                // We have to keep trying to activate this node with the
+                // matching ingress key until it becomes active.
+                //
+                // This prevents the secenario in which the active node is
+                // bounced, and then we end up here and we're trying to
+                // activate an idle node. If we give up after a few
+                // unsuccessful tries, then we'd report the previously
+                // active key as lost. But then the previously active server
+                // could be restarted by Kubernetes, have this lost key, and
+                // restart in the active state. This would be bad because
+                // we don't want a server to be active with a lost key.
+                //
+                // Since we always expect the node to come back in some
+                // time, we should just keep trying here.
+                //
+                // TODO: Add alerting if this is taking too long.
+                loop {
+                    log::info!(
+                        self.logger,
+                        "Trying to activate node at index {}",
+                        ingest_summary_node_mapping.node_index
+                    );
+                    match self.ingest_clients[ingest_summary_node_mapping.node_index].activate() {
+                        Ok(_) => {
+                            log::info!(self.logger, "Successfully activated ingest client.");
+                            return Ok(());
+                        }
+                        Err(_) => {
+                            // TODO: See if we can make an id field for ingest client. This
+                            // will allow us to better identify ingest clients.
+                            log::warn!(self.logger, "Could not activate idle ingest client.");
+                        }
+                    }
+                }
+            }
+        }
+
+        Err(OverseerError::ActivateNode("Could not activate a node because no node has an ingress key that matches the outstanding key.".to_string()))
+    }
+
+    fn report_lost_ingress_key(
+        &self,
+        inactive_outstanding_key: CompressedRistrettoPublic,
+    ) -> Result<(), OverseerError> {
+        let result = retry_with_index(Fixed::from_millis(200), |current_try| {
+            if current_try as u8 > Self::NUMBER_OF_TRIES {
```

Comment:
I used it because I'd like the error message to include how many tries remain. I don't think the take option enables this? If it does, then great- it's more compact and would change to it.

Relevant [documentation](https://docs.rs/retry/latest/retry/#usage):
`If your operation needs to know how many times it’s been tried, use the retry_with_index function. This works the same as retry, but passes the number of the current try to the closure as an argument.`

Reply:
You can use a combination of `retry_with_index` and `take`, which will allow you to include the number of retries remaining. It will save you from having to implement the limiting yourself.